components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1528,8 +1531,35 @@ void RewardsServiceImpl::GetAddresses(const GetAddressesCallback& callback) {
     return;
   }
 
-  bat_ledger_->GetAddresses(base::BindOnce(&RewardsServiceImpl::OnGetAddresses,
-        AsWeakPtr(), callback));
+  bat_ledger_->GetAddressesLimitedCountries(
```

Comment:
This logic needs to be moved to ledger code. 

So what should happen is that we send country code to ledger and ledger return addresses. Rewards service doesn't need to know about limitations. We shouldn't have business logic in here.

Reply:
I agree but I wasn't sure if I should call country_codes::CountryCharsToCountryID in ledger.

Reply:
So instead, I'll move to ledger and make the call to `country_codes::CountryCharsToCountryID` through `LedgerClient`

Reply:
what I would do is just pass country into `GetAddresses` function so that ledger can know what is your current country and then ledger can take it from there

Reply:
this way ledger don't need to do another call out

Reply:
What's happening here is the call to `country_codes::GetCountryIdFromPrefs` and `country_codes::CountryCharsToCountryID` take the characters and packs them into an integer (US is something like `20154` JP is something like `19345`) and the current country ID (5 digit number) is what is stored in prefs. So to get the comparison, we have to do another call out to get the ID to compare.

Reply:
we could send in country ID and instead of storing JP in ledger we store ID (which is ISO 3166-1 country code)

Reply:
No, the ISO 3166-1 code is a 3 digit code. 

The code that Cr is storing in prefs is a code that is converted and packed based off of the alpha-2 code received. The result is a 5 digit code.

https://chromium.googlesource.com/chromium/src/+/refs/heads/master/components/country_codes/country_codes.h#7

Reply:
Logic is moved to Ledger. There is a callout to LedgerClient to get the Cr specific country code from the alpha-2 chars.